dags/process_elife_zip_dag.py

```diff
@@ -86,62 +146,31 @@ def extract_archived_files_to_bucket(**context) -> str:
 
         prefix = zip_file_name.replace('.zip', '/')
 
+        args = [(prefix, temp_zip_file.name, f) for f in ZipFile(temp_zip_file).namelist()]
+
         # extract zip to cloud bucket
-        for zipped_file_path in ZipFile(temp_zip_file).namelist():
-            # extract zipped files to disk to avoid using too much memory
-            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_unzipped_file:
-                temp_unzipped_file.write(ZipFile(temp_zip_file).read(zipped_file_path))
-                temp_unzipped_file.seek(0)
-
-                s3_key = prefix + zipped_file_path
-                s3.put_object(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=s3_key,
-                    Body=temp_unzipped_file
-                )
-                logger.info('%s uploaded to %s', s3_key, DESTINATION_BUCKET)
+        with ThreadPoolExecutor() as p:
+            uploaded_files = p.map(extract_file_to_s3, args)
 
         # check if expected article in zip file
         if not [fn for fn in ZipFile(temp_zip_file).namelist() if article_name in fn]:
             error_message = '%s not in %s: %s' % (article_name, zip_file_name,
                                                   ZipFile(temp_zip_file).namelist())
             raise FileNotFoundError(error_message)
 
-    # pass article cloud bucket key to next task
-    return '%s/%s' % (prefix, article_name)
+    return list(uploaded_files)
 
 
-def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> None:
+def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> List[str]:
     zip_file_name = get_file_name_passed_to_dag_run_conf_file(context)
     prefix = zip_file_name.replace('.zip', '/')
+    tiffs = {key
+             for key in list_bucket_keys_iter(Bucket=DESTINATION_BUCKET, Prefix=prefix)
+             if key.endswith('.tif')}
 
-    s3 = get_s3_client()
-
-    for key in list_bucket_keys_iter(Bucket=DESTINATION_BUCKET, Prefix=prefix):
-        if key.endswith('.tif'):
-            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_tiff_file:
-                s3.download_fileobj(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=key,
-                    Fileobj=temp_tiff_file
-                )
-                temp_tiff_file.seek(0)
-
-                logger.info('converting %s to jpeg', key)
-
-                temp_jpeg = BytesIO()
-                with Image(file=temp_tiff_file) as img:
-                    img.format = 'jpeg'
-                    img.save(file=temp_jpeg)
-
-                key = re.sub(r'\.\w+$', '.jpg', key)
-                s3.put_object(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=key,
-                    Body=temp_jpeg.getvalue()
-                )
-                logger.info('%s uploaded to %s',key, DESTINATION_BUCKET)
-
+    with ThreadPoolExecutor() as p:
```

Comment:
same here, can be a global configuration rather than being specific for the task

Reply:
updated.